src/test/index.test.ts

```diff
@@ -0,0 +1,164 @@
+
+const request = require('supertest')
+const express = require('express')
+const path = require('path')
+const fs = require('fs')
+let { env } = require('node:process')
+
+const metricsConfig: MetricsConfigurationOptions = {
+  defaultMetricsLabels: {
+    one: 'test1',
+    two: 'test2',
+    three: 'test3',
+  },
+}
+
+jest.mock('node:http', () => ({
+  get: (_: any, callback: any) => {
+    const response = {
+      on: (event: string, handler: any) => {
+        if (event === 'data') {
+          const filePath = path.resolve(__dirname, '../../ecs_metadata/container.json')
+          const mockData = fs.readFileSync(filePath, 'utf8')
+          handler(Buffer.from(mockData))
+        } else if (event === 'end') {
+          handler()
+        }
+      },
+      end: () => {},
+    }
+    return callback(response)
+  },
+}))
+describe('/metrics endpoint', () => {
+  const envRestorePoint = env
+  let app: Express
+  let metrics = require('../index')
+
+  beforeEach(() => {
+    env = envRestorePoint
+    jest.resetModules()
+    app = express()
+    metrics = require('../index')
+  })
+
+  it('should return metrics as plain text by default', async () => {
+    app.use(metrics.configure(metricsConfig))
+    const response = await request(app).get('/metrics')
+    expect(response.status).toBe(200)
+    expect(response.headers['content-type']).toContain('text/plain')
+    expect(response.text).toContain('nodejs_eventloop_lag_stddev_seconds')
+    expect(response.text).toContain('nodejs_gc_duration_seconds')
+    expect(response.text).toContain('process_resident_memory_bytes')
+  })
+
+  it('should return metrics as json when Accept header is set', async () => {
+    app.use(metrics.configure(metricsConfig))
+    const response = await request(app).get('/metrics').set('Accept', 'application/json')
+    let jsonArray = [{}]
+    expect(response.status).toBe(200)
+    expect(response.headers['content-type']).toContain('application/json')
+    expect(() => (jsonArray = JSON.parse(response.text))).not.toThrow()
+
+    const expectedKeyValuePairs = [
+      { name: 'process_cpu_user_seconds_total' },
+      { name: 'nodejs_active_resources' },
+      { name: 'nodejs_heap_space_size_available_bytes' },
+    ]
+
+    expectedKeyValuePairs.forEach((pair) => {
+      expect(jsonArray).toContainEqual(expect.objectContaining(pair))
+    })
+  })
+
+  it('should return custom metrics with default and custom labels', async () => {
+    app.use(metrics.configure(metricsConfig))
+    const test_counter = metrics.registerCounter('test_counter', 'test counter metric', ['test_type']) // user defined metric label 'test_type'
+    const test_gauge = metrics.registerGauge('test_gauge', 'test gauge metric', ['test_type']) // user defined metric label 'test_type'
+    const test_histogram = metrics.registerHistogram('test_histogram', 'test histogram metric', ['test_type']) // user defined metric label 'test_type'
```

Comment:
Do we need the `// user defined` comments here?

Reply:
nope, I had these comments in the demo code as an explainer. They must have been copied over by accident when I was writing the tests